### ISSUE TITLE
Bug: ESLint 8 breaks `indent` rule when using decorators

### ISSUE THREAD

**Issue by newcat** [eslint/eslint#15299]
### Environment

Node version: v14.17.6
npm version: v6.14.15
Local ESLint version: v8.2.0 (Currently used)
Global ESLint version: Not found
Operating System: win32 10.0.19042

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```
{
  "root": true,
  "plugins": ["@typescript-eslint"],
  "extends": ["eslint:recommended", "plugin:@typescript-eslint/recommended", "prettier"],
  "parser": "vue-eslint-parser",
  "parserOptions": {
    "ecmaVersion": 12,
    "parser": "@typescript-eslint/parser",
    "sourceType": "module",
    "project": "tsconfig.json",
    "extraFileExtensions": [".vue"]
  },
  "rules": {
    "max-len": ["error", 120, { "ignoreTemplateLiterals": true }],
    "indent": ["error", 2, { "SwitchCase": 1 }],
    "quotes": ["error", "double", { "avoidEscape": true }],
    "semi": ["error", "always"],
    "@typescript-eslint/no-for-in-array": "error",
    "@typescript-eslint/no-floating-promises": "error",
    "@typescript-eslint/await-thenable": "error",
    "@typescript-eslint/no-misused-promises": "error",
    "require-await": "off",
    "@typescript-eslint/require-await": "error",
    "@typescript-eslint/explicit-module-boundary-types": "off",
    "@typescript-eslint/no-non-null-assertion": "off"
  }
}

```
</details>

```ts
import { Prop, Schema, SchemaFactory } from "@nestjs/mongoose";
import * as mongoose from "mongoose";

@Schema()
export class Foo {
  @Prop()
  bar!: string;
}
```

### What did you expect to happen?

No indentation error, as it was in ESLint 7.32.0

### What actually happened?

```
7:1  error  Expected indentation of 4 spaces but found 2  indent
```

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

I tried with ESLint versions 7.32.0, 8.0.0 and 8.2.0.
With 7.32.0 I don't get these errors - but just updating ESLint to 8.0.0 (didn't change anything else) produces the errors.

**Comment by nzakas** [comment]
Thanks for the report. The ESLint core doesn’t officially support TypeScript or decorators, so I’m not sure why this was working previously or why it wouldn’t work now. It also doesn’t look like there were any changes to the rule during the release.

Is it possible that typescript-eslint changed in some way? @bradzacher

**Comment by bradzacher** [comment]
Nope - it's a really poorly done extension rule that was written badly to begin with (by me) and hasn't aged well at all.

TL;DR - it's unsupported, and known to be broken. Nobody in the community has been motivated enough to work to fix it. https://github.com/typescript-eslint/typescript-eslint/issues/1824

Likely you guys made a change to your base code in v8 which subtly influenced the hacks that are implemented. Not your fault at all. Feel free to close this out.

**Comment by mdjermanovic** [comment]
> ```ts
>   @Prop()
>   bar!: string;
> ```

@newcat does ESLint v7.32.0 enforce any indentation on these lines, or does it just ignore them?

In the latter case, you can restore this behavior in ESLint 8 by setting `"ignoredNodes": ["PropertyDefinition"]` option:

```json
"indent": ["error", 2, { "SwitchCase": 1, "ignoredNodes": ["PropertyDefinition"] }]
```

**Comment by newcat** [comment]
@mdjermanovic thanks for the hint. Indeed, v7.32.0 does not enforce the indentation in the example - that's why there were no errors in that version 😄 

The workaround does have the unwanted side effect that the property definitions in every class are affected, independent of whether they have a decorator or not. But I guess that's better than having a lot of false errors and also this should be taken care of by Prettier anyway. So thanks for the hint 👍 

Will close this issue now, since the reason is likely what [Brad described](https://github.com/eslint/eslint/issues/15299#issuecomment-967762181)
